sdk/storage/azblob/service/models.go

```diff
@@ -21,6 +21,22 @@ func NewSharedKeyCredential(accountName, accountKey string) (*SharedKeyCredentia
 	return exported.NewSharedKeyCredential(accountName, accountKey)
 }
 
+// UserDelegationCredential contains an account's name and its user delegation key.
+type UserDelegationCredential = exported.UserDelegationCredential
+
+// UserDelegationKey contains UserDelegationKey.
+type UserDelegationKey = generated.UserDelegationKey
+
+type KeyInfo = generated.KeyInfo
+
+// NewUserDelegationCredential creates a new UserDelegationCredential using a Storage account's name and a user delegation key from it
+func NewUserDelegationCredential(accountName string, udk UserDelegationKey) *UserDelegationCredential {
+	return exported.NewUserDelegationCredential(accountName, udk)
+}
+
+// GetUserDelegationKeyOptions contains request id and timeout
+type GetUserDelegationKeyOptions = generated.ServiceClientGetUserDelegationKeyOptions
```

Comment:
I don't think we want to expose this underlying options type as all it contains is `RequestID` and `Timeout` fields which we hide from customers.  Probably best to define an empty options type for now (see `GetPropertiesOptions` in this file for the pattern).